ansible/roles/test/files/ptftests/arista.py

```diff
@@ -313,6 +331,49 @@ def parse_bgp_route(self, output, expects):
 
         return set(expects) == prefixes
 
+    def get_bgp_neighbors(self):
+        neigh_bgp = {}
+        dut_bgp = {}
+        for cmd, ver in [('show ip bgp neighbors', 'v4'), ('show ipv6 bgp neighbors', 'v6')]:
+            output = self.do_cmd(cmd)
+            if ver == 'v6':
+                neigh_bgp[ver], dut_bgp[ver], neigh_bgp['asn'] = self.parse_bgp_neighbor(output, get_neigh=True, get_asn=True)
+            else:
+                neigh_bgp[ver], dut_bgp[ver] = self.parse_bgp_neighbor(output, get_neigh=True)
+
+        return neigh_bgp, dut_bgp
+
+    def change_bgp_neigh_state(self, asn, state="no shut"):
+        self.do_cmd('configure')
+        self.do_cmd("router bgp %s" % asn)
+        self.do_cmd("%s" % state)
+        self.do_cmd('exit')
+        self.do_cmd('exit')
+
+    def verify_bgp_neigh_state(self, dut=None, state="Active"):
+        bgp_state = {}
+        bgp_state['v4'] = bgp_state['v6'] = False
+        for cmd, ver in [('show ip bgp summary | json', 'v4'), ('show ipv6 bgp summary | json', 'v6')]:
+            output = self.do_cmd(cmd)
+            data = "\n".join(output.split("\r\n")[1:-1])
+            obj = json.loads(data)
+
+            if state != 'Active':
+               if 'vrfs' in obj:
+                   bgp_state[ver] = not obj['vrfs']
```

Comment:
>not obj['vrfs'] [](start = 36, length = 15)

What is the intention of this line?

Reply:
When I shut down the bgp neighbor on the Arista end, the state is 'down' and obj['vrfs'] is empty. bgp_state[ver] is a bool here. So in this case it will return True which is the what I expect.

Reply:
Then you can share this info as code comment.

---
In reply to: [287447750](https://github.com/Azure/sonic-mgmt/pull/926#discussion_r287447750) [](ancestors = 287447750)